### PR TITLE
fix: port fallback logic in domain creator

### PR DESCRIPTION
**Pull request by anshumanv**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case
Fix logic

### Breaking Changes
No

### CODE REVIEW

File: lib/utils/createDomain.js [hunk 1]
```diff
@@ -17,7 +18,7 @@ function createDomain(options, server) {
     hostname = '0.0.0.0';
   }
 
-  const port = server ? server.address().port : 0;
+  const port = server ? server.address().port : defaultPort;
```
Comment (ylemkimon) [inline lib/utils/createDomain.js:21]: The intention was for client to use the `location` port: https://github.com/webpack/webpack-dev-server/blob/723e1ec394d2f1bdeda2e80dc8748b6da1c0a3a3/client-src/default/utils/createSocketUrl.js#L48-L50

Comment (anshumanv) [reply lib/utils/createDomain.js:21]: hmm yes missed this, @alexander-akait  better alternate?

Comment (alexander-akait) [reply lib/utils/createDomain.js:21]: oh, good catch, we need refactor logic here, it is more complex

**Comment by alexander-akait** [comment]
Fixed in master

**Pull request closed without merging**